wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1202,6 +1204,41 @@ instance BuildableSafeGen AddressValidity where
     buildSafeGen _ AddressValidity{..} =
         bprint ("{ valid="%build%" }") isValid
 
+-- | An address is either recognised as "ours" or not. An address that is not
+--   recognised may still be ours e.g. an address generated by another wallet instance
+--   will not be considered "ours" until the relevant transaction is confirmed.
+--
+--   In other words, `AddressNotRecognised` makes an inconclusive statement about
+--   an address, whereas `AddressIsOurs` is unambiguous.
+data AddressIsOurs = AddressIsOurs
+                   | AddressNotRecognised
+    deriving (Show, Eq, Generic, Ord)
+
+addressIsOursFromUsed :: Bool -> V1 AddressIsOurs
+addressIsOursFromUsed used =
+    V1 $ if used then AddressIsOurs else AddressNotRecognised
+
+instance ToJSON (V1 AddressIsOurs) where
+    toJSON (V1 AddressIsOurs)        = String "is-ours"
+    toJSON (V1 AddressNotRecognised) = String "not-recognised"
+
+instance FromJSON (V1 AddressIsOurs) where
+    parseJSON (String "is-ours")        = pure (V1 AddressIsOurs)
+    parseJSON (String "not-recognised") = pure (V1 AddressNotRecognised)
+    parseJSON x = typeMismatch "Not a valid AddressIsOurs" x
```

Comment:
If you haven't done so (haven't reached to that point of the PR perhaps) don't forget to write a roundtrip Aeson test inside the `MarshallingSpecs`!

Reply:
Same remarks as above, what about leveraging generics here?